src/storage/store/engine.rs

```diff
@@ -536,61 +536,76 @@ impl ShardEngine {
                 }
             }
             if let Some(fname_transfer) = &msg.fname_transfer {
-                if fname_transfer.proof.is_none() {
-                    warn!(
-                        fid = snapchain_txn.fid,
-                        id = fname_transfer.id,
-                        "Fname transfer has no proof"
-                    );
-                }
-
-                match verification::validate_fname_transfer(fname_transfer) {
-                    Ok(_) => {}
-                    Err(err) => {
-                        warn!("Error validating fname transfer: {:?}", err);
-                    }
-                }
-
-                let event = UserDataStore::merge_username_proof(
-                    &self.stores.user_data_store,
-                    fname_transfer.proof.as_ref().unwrap(),
-                    txn_batch,
-                );
-                match event {
-                    Ok(hub_event) => {
-                        merged_fnames_count += 1;
-                        self.update_trie(&merkle_trie::Context::new(), &hub_event, txn_batch)?;
-                        events.push(hub_event.clone());
-                        system_messages_count += 1;
-                    }
-                    Err(err) => {
-                        warn!("Error merging fname transfer: {:?}", err);
+                match &fname_transfer.proof {
+                    None => {
+                        warn!(
+                            fid = snapchain_txn.fid,
+                            id = fname_transfer.id,
+                            "Fname transfer has no proof"
+                        );
+                        return Err(EngineError::EngineMessageValidationError(
```

Comment:
I don't think we should return here because that would skip processing the rest of the messages in the transaction. 

We should probably break up this function into smaller functions so it's easier to reason about and we can't make this kind of mistake. I'll do this in a separate PR.